src/io/functions/iologindata_save_player.cpp

```diff
@@ -243,234 +243,209 @@ bool IOLoginDataSave::savePlayerFirst(const std::shared_ptr<Player> &player) {
+	columns.push_back(fmt::format("`skill_critical_hit_damage` = {}", player->skills[SKILL_CRITICAL_HIT_DAMAGE].level));
+	columns.push_back(fmt::format("`skill_critical_hit_damage_tries` = {}", player->skills[SKILL_CRITICAL_HIT_DAMAGE].tries));
+	columns.push_back(fmt::format("`skill_life_leech_chance` = {}", player->skills[SKILL_LIFE_LEECH_CHANCE].level));
+	columns.push_back(fmt::format("`skill_life_leech_chance_tries` = {}", player->skills[SKILL_LIFE_LEECH_CHANCE].tries));
+	columns.push_back(fmt::format("`skill_life_leech_amount` = {}", player->skills[SKILL_LIFE_LEECH_AMOUNT].level));
+	columns.push_back(fmt::format("`skill_life_leech_amount_tries` = {}", player->skills[SKILL_LIFE_LEECH_AMOUNT].tries));
+	columns.push_back(fmt::format("`skill_mana_leech_chance` = {}", player->skills[SKILL_MANA_LEECH_CHANCE].level));
+	columns.push_back(fmt::format("`skill_mana_leech_chance_tries` = {}", player->skills[SKILL_MANA_LEECH_CHANCE].tries));
+	columns.push_back(fmt::format("`skill_mana_leech_amount` = {}", player->skills[SKILL_MANA_LEECH_AMOUNT].level));
+	columns.push_back(fmt::format("`skill_mana_leech_amount_tries` = {}", player->skills[SKILL_MANA_LEECH_AMOUNT].tries));
+	columns.push_back(fmt::format("`stamina` = {}", player->getStaminaMinutes()));
+	columns.push_back(fmt::format("`manashield` = {}", player->getManaShield()));
+	columns.push_back(fmt::format("`max_manashield` = {}", player->getMaxManaShield()));
+	columns.push_back(fmt::format("`xpboost_value` = {}", player->getXpBoostPercent()));
+	columns.push_back(fmt::format("`xpboost_stamina` = {}", player->getXpBoostTime()));
+	columns.push_back(fmt::format("`quickloot_fallback` = {}", player->quickLootFallbackToMainContainer ? 1 : 0));
+
+	// Blessings
+	for (int i = 1; i <= 8; ++i) {
+		columns.push_back(fmt::format("`blessings{}` = {}", i, static_cast<uint32_t>(player->getBlessingCount(static_cast<uint8_t>(i)))));
+	}
+
+	// Create the task for the query execution
+	auto savePlayerTask = [joinColums = std::move(columns), playerName = player->getName(), playerGUID = player->getGUID()]() {
+		// Now join the columns into a single string
+		std::string setClause = fmt::to_string(fmt::join(joinColums, ", "));
+		// Construct the final query
+		std::string queryStr = fmt::format("UPDATE `players` SET {} WHERE `id` = {}", setClause, playerGUID);
+		if (!g_database().executeQuery(queryStr)) {
+			g_logger().warn("[SaveManager::savePlayerTask] - Error executing player first save for player: {}", playerName);
+		}
+	};
 
-	if (!stashQuery.execute()) {
-		return false;
-	}
+	// Add the task to the SaveManager
+	g_saveManager().addTask(savePlayerTask, "IOLoginData::savePlayerFirst - savePlayerTask");
 	return true;
 }
 
 bool IOLoginDataSave::savePlayerSpells(const std::shared_ptr<Player> &player) {
 	if (!player) {
-		g_logger().warn("[IOLoginData::savePlayer] - Player nullptr: {}", __FUNCTION__);
+		g_logger().warn("[IOLoginDataSave::savePlayerSpells] - Player nullptr: {}", __FUNCTION__);
```

Comment:
To increase the manutenability and avoid renaming every method name update, also there is no need to pass the function name twice.
```suggestion
		g_logger().warn("[{}] - Player nullptr", std::source_location::current().function_name());
```